t/t2606-job-shell-output-redirection.t

```diff
@@ -0,0 +1,447 @@
+#!/bin/sh
+#
+test_description='Test flux-shell'
+
+. `dirname $0`/sharness.sh
+
+jq=$(which jq 2>/dev/null)
+test -n "$jq" && test_set_prereq HAVE_JQ
+
+test_under_flux 4 job
+
+flux setattr log-stderr-level 1
+
+TEST_SUBPROCESS_DIR=${FLUX_BUILD_DIR}/src/common/libsubprocess
+
+hwloc_fake_config='{"0-3":{"Core":2,"cpuset":"0-1"}}'
+
+test_expect_success 'job-shell: load barrier,job-exec,sched-simple modules' '
+        #  Add fake by_rank configuration to kvs:
+        flux kvs put resource.hwloc.by_rank="$hwloc_fake_config" &&
+        flux module load barrier &&
+        flux module load -r 0 sched-simple &&
+        flux module load -r 0 job-exec
+'
+
+test_expect_success 'job-shell: generate 1-task echo jobspecs' '
+        flux jobspec srun -N1 -n1 ${TEST_SUBPROCESS_DIR}/test_echo -P -O foo > j1echostdout &&
+        flux jobspec srun -N1 -n1 ${TEST_SUBPROCESS_DIR}/test_echo -P -E bar > j1echostderr &&
+        flux jobspec srun -N1 -n1 ${TEST_SUBPROCESS_DIR}/test_echo -P -O -E baz > j1echoboth
+'
+
+test_expect_success 'job-shell: generate 2-task echo jobspecs' '
+        flux jobspec srun -N1 -n2 ${TEST_SUBPROCESS_DIR}/test_echo -P -O foo > j2echostdout &&
+        flux jobspec srun -N1 -n2 ${TEST_SUBPROCESS_DIR}/test_echo -P -E bar > j2echostderr &&
+        flux jobspec srun -N1 -n2 ${TEST_SUBPROCESS_DIR}/test_echo -P -O -E baz > j2echoboth
+'
+
+#
+# 1 task output file tests
+#
+
+test_expect_success HAVE_JQ 'job-shell: run 1-task echo job (stdout file)' '
+        cat j1echostdout \
+            |  $jq ".attributes.system.shell.options.output.stdout.type = \"file\"" \
+            |  $jq ".attributes.system.shell.options.output.stdout.path = \"out0\"" \
+            > j1echostdout-0 &&
+        id=$(cat j1echostdout-0 | flux job submit) &&
+        flux job wait-event $id clean &&
+        grep stdout:foo out0
+'
```

Comment:
Now that `flux mini run` has been merged, and has options for file redirection, I think many of these tests can be shortened and HAVE_JQ dropped, e.g. the above could be
```shell
flux mini run -n1 --output=out0 ${TEST_SUBPROCESS_DIR}/test_echo -P -O foo &&
grep stdout:foo out0
```
In addition to `--output` and `--error`, there is also `-o, --setopt` to directly set shell option attributes.